projects/ng-wizard/src/lib/ng-wizard.service.ts

```diff
@@ -77,15 +78,15 @@ export class NgWizardService {
     // strategy.
     // The path is based on the current route to allow route parameter
     const routeFragment = this.router.url.split('/');
-    routeFragment[routeFragment.length - 1] = stepData.path;
+    routeFragment.pop();
+    routeFragment.push(stepData.path);
     const stepPath = routeFragment.join('/');
 
     if (stepData.options.cleanQueryParameters) {
       return this.router.navigate([stepPath], { queryParams: {} });
     } else {
       return this.router.navigate([stepPath], { queryParamsHandling: 'merge' });
```

Comment:
just a cosmetic change: the value being returned does not need to be in the 'else'